src/Parser/DeathKnight/Blood/Modules/Talents/Bonestorm.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage, formatNumber } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Wrapper from 'common/Wrapper';
+
+class Bonestorm extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  bsCasts = [];
+  cast = 0;
+  totalBonestormDamage = 0;
+  totalHits = 0;
+  totalCost = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.BONESTORM_TALENT.id);
+  }
+
+  on_byPlayer_cast(event) {
+    if (event.ability.guid !== SPELLS.BONESTORM_TALENT.id) {
+      return;
+    }
+
+    this.bsCasts.push({
+      timestamp: event.timestamp,
+      expire: event.timestamp + (event.classResources[0].cost * 10),
+      cost: event.classResources[0].cost,
+      hits: [],
+      cast: this.cast,
```

Comment:
You could drop this prop since it's always equal to the index and just use the index in it's place.